cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -1451,6 +1451,60 @@ public void testCruiseControlDisabledToEnabledBehaviour(VertxTestContext context
                 }));
     }
 
+    @Test
+    public void testKrWhenCCUserTaskIsEmpty(VertxTestContext context) throws IOException, URISyntaxException {
+
+        cruiseControlServer.setupCCRebalanceResponse(0, CruiseControlEndpoints.REBALANCE);
+
+        KafkaRebalance kr = new KafkaRebalanceBuilder(createKafkaRebalance(namespace, CLUSTER_NAME, RESOURCE_NAME, EMPTY_KAFKA_REBALANCE_SPEC, false))
+                .withNewStatus()
+                .withObservedGeneration(1L)
+                .withConditions(new ConditionBuilder()
+                        .withType("ProposalReady")
+                        .withStatus("True")
+                        .build())
+                .endStatus()
+                .build();
+
+        Crds.kafkaRebalanceOperation(client).inNamespace(namespace).resource(kr).create();
+        Crds.kafkaRebalanceOperation(client).inNamespace(namespace).resource(kr).updateStatus();
+
+        crdCreateKafka();
+        crdCreateCruiseControlSecrets();
+
+        Checkpoint checkpoint = context.checkpoint();
+        krao.reconcile(new Reconciliation("test-trigger", KafkaRebalance.RESOURCE_KIND, namespace, RESOURCE_NAME))
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+                    // the resource moved to ProposalReady
+                    assertState(context, client, namespace, RESOURCE_NAME, KafkaRebalanceState.ProposalReady);
```

Comment:
As discussed in the other thread - you are already creating it as ProposalReady. So it should not move to ProposalReady here. so the comment is wrong ... and it is not clear what the first reconciliation really does.

Reply:
The first reconciliation doesn't do much, it is just a normal reconcile after which Iam checking if the `proposal` was in the correct state i.e. `ProposalReady` but yah I can remove this extra step.

Reply:
Normally, in a unit test, you would try to avoid doing things that _doesn't do much_. It either does something that matters to the test and that is why it is there ... or it can be removed. 😉

Reply:
Removed the unnecessary steps